OpenQA/Isotovideo/Utils.pm

```diff
@@ -37,6 +37,42 @@ sub calculate_git_hash ($git_repo_dir) {
     return $git_hash;
 }
 
+sub clone_git ($local_path, $clone_url, $clone_depth, $branch, $dir, $dir_variable) {
+    return bmwqemu::diag "Skipping to clone '$clone_url'; $local_path already exists" if -e $local_path;
+    bmwqemu::fctinfo "Cloning git URL '$clone_url'";
+    my $branch_args = '';
+    if ($branch) {
+        bmwqemu::fctinfo "Checking out git refspec/branch '$branch'";
+        $branch_args = " --branch $branch";
+    }
+    my $clone_cmd = 'env GIT_SSH_COMMAND="ssh -oBatchMode=yes" git clone';
+    my @out = qx{$clone_cmd --depth=$clone_depth $branch_args $clone_url 2>&1};
+    my $handle_output = sub ($return_value, @out) {
+        bmwqemu::diag "@out" if @out;
+        die "Unable to clone Git repository '$dir' specified via $dir_variable (see log for details)" unless $return_value == 0;
+        return 1;
+    };
+    return $handle_output->($?, @out) unless ($branch && grep /warning: Could not find remote branch/, @out);
+    # maybe we misspelled or maybe someone gave a commit hash instead
+    # for which we need to take a different approach by downloading the
+    # repository in the necessary depth until we can reach the commit
+    # References:
+    # * https://stackoverflow.com/questions/18515488/how-to-check-if-the-commit-exists-in-a-git-repository-by-its-sha-1
+    # * https://stackoverflow.com/questions/26135216/why-isnt-there-a-git-clone-specific-commit-option
+    bmwqemu::diag "Fetching more remote objects to ensure availability of '$branch'";
+    @out = qx{$clone_cmd --depth=$clone_depth $clone_url 2>&1};
+    $handle_output->($?, @out);
+    while (qx[git -C $local_path cat-file -e $branch^{commit} 2>&1] =~ /Not a valid object/) {
+        $clone_depth *= 2;
+        @out = qx[git -C $local_path fetch --progress --depth=$clone_depth 2>&1];
+        $handle_output->($?, @out);
+        die "Could not find '$branch' in complete history in cloned Git repository '$dir'" if grep /remote: Total 0/, @out;
+    }
+    qx{git -C $local_path checkout $branch};
```

Comment:
Maybe it would be good to save the output and print it in case of failure

Reply:
Actually we should probably just always print it. That's what we do in other cases whether it fails or not.